src/test/neorados/CMakeLists.txt

```diff
@@ -21,8 +20,184 @@ target_link_libraries(ceph_test_neorados_op_speed
 
 add_library(neoradostest-support STATIC common_tests.cc)
 target_link_libraries(neoradostest-support
-  libneorados fmt::fmt)
+  libneorados fmt::fmt GTest::GTest)
 
 add_executable(ceph_test_neorados_list_pool list_pool.cc)
 target_link_libraries(ceph_test_neorados_list_pool
   libneorados neoradostest-support global fmt::fmt ${unittest_libs})
+
+add_executable(ceph_test_neorados_handler_error
```

Comment:
i'd like to make sure these new test cases run somewhere in the rados suite. there's a `qa/workunits/rados/test.sh` that already runs several `ceph_test_rados_*` tests, maybe that would be a good place to add these? that workunit is part of several subsuites:
```
~/ceph $ grep -r rados/test.sh qa/suites/rados
qa/suites/rados/verify/tasks/rados_api_tests.yaml:        - rados/test.sh
qa/suites/rados/monthrash/workloads/rados_api_tests.yaml:        - rados/test.sh
qa/suites/rados/basic/tasks/rados_api_tests.yaml:        - rados/test.sh
qa/suites/rados/thrash/workloads/rados_api_tests.yaml:        - rados/test.sh
```